service/storage_service.cc

```diff
@@ -354,6 +354,13 @@ future<> storage_service::topology_state_load(cdc::generation_service& cdc_gen_s
             co_await add_normal_node(id, rs);
         }
 
+        {
+            const auto stage = _topology_state_machine._topology.tstate == topology::transition_state::write_both_read_new
+                ? locator::topology_change_stage::read_new
+                : locator::topology_change_stage::read_old;
+            tmptr->set_topology_change_stage(stage);
```

Comment:
Here it will simply be `tmptr->set_topology_change_stage(_topology_state_machine._topology.tstate)`. At some point we will need to merge `topology_metadata` and `topology` classes.

Reply:
> At some point we will need to merge topology_metadata and topology classes?

Why?

To me it makes as much sense as merging `gossiper::_endpoint_state_map` with `token_metadata`.

Reply:
> Why?

They largely duplicate one another. Topology has some data that tm does not, but it can be stored in new `node` structure now. Well, may be what we really need to unify is `service::topology` and `locator::topology` but the `token_metadata` is closely coupled with the later anyway.